### PR TITLE
[4] fixed undefined/wrong exception classname in com_privacy

### PR DESCRIPTION
**Pull request by PhilETaylor**
Code review

### CODE REVIEW

File: administrator/components/com_privacy/src/View/Export/XmlView.php [hunk 1]
```diff
@@ -44,7 +45,7 @@ public function display($tpl = null)
 		// Check for errors.
 		if (count($errors = $this->get('Errors')))
 		{
-			throw new \JViewGenericdataexception(implode("\n", $errors), 500);
+			throw new \GenericDataException(implode("\n", $errors), 500);
```
Comment (joomdonation) [inline administrator/components/com_privacy/src/View/Export/XmlView.php]: Typo. Should be 

```php
throw new GenericDataException(implode("\n", $errors), 500);
```

instead.

**Comment by brianteeman** [comment]
I'm sure I was told previously that these should not be changed<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34005">issues.joomla.org/tracker/joomla-cms/34005</a>.</sub>

File: administrator/components/com_privacy/src/View/Export/XmlView.php (hunk 1 above)
Comment (PhilETaylor) [reply administrator/components/com_privacy/src/View/Export/XmlView.php]: Thanks. I was sleepy by this time.

**Comment by PhilETaylor** [comment]
@brianteeman that might have been the case at the time, whenever that was, but now this class doesn't exist and there is no alias for it mapping it to the non J prefixed correct class.

**Comment by brianteeman** [comment]
there are quite a few other ones to change then ;)

**Comment by PhilETaylor** [comment]
> there are quite a few other ones to change then ;)

Such as? or do you simply mean the ones in https://github.com/joomla/joomla-cms/pull/34005/commits/9eaaccd4a4c0d389b16d612c2d1cc77104b5d113 that I did not push from my bed last night but did before your comment/

File: components/com_privacy/src/View/Remind/HtmlView.php [hunk 2]
```diff
@@ -11,9 +11,11 @@
 
 \defined('_JEXEC') or die;
 
+use Exception;
```
Comment (joomdonation) [inline components/com_privacy/src/View/Remind/HtmlView.php]: No use statement place, change `@throws  \Exception` in docblock instead.

File: components/com_privacy/src/View/Request/HtmlView.php [hunk 3]
```diff
@@ -11,9 +11,11 @@
 
 \defined('_JEXEC') or die;
 
+use Exception;
```
Comment (joomdonation) [inline components/com_privacy/src/View/Request/HtmlView.php]: No use statement place, change `@throws  \Exception` in docblock instead.

**Comment by PhilETaylor** [comment]
Doen.

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on c50f3da23849dfa0afb3af1ac0f621ddf0278fe3<br /><br />Many thanks !<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34005">issues.joomla.org/tracker/joomla-cms/34005</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on c50f3da23849dfa0afb3af1ac0f621ddf0278fe3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34005">issues.joomla.org/tracker/joomla-cms/34005</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34005">issues.joomla.org/tracker/joomla-cms/34005</a>.</sub>

**Pull request merged**